src/savers/tvg/tvgTvgSaver.cpp

```diff
@@ -200,12 +200,59 @@ TvgBinCounter TvgSaver::serializePaint(const Paint* paint)
 }
 
 
+/* Propagate parents properties to the child so that we can skip saving the parent. */
+TvgBinCounter TvgSaver::serializeChild(const Paint* parent, const Paint* child, const Matrix* transform)
+{
+    const Paint* compTarget = nullptr;
+    auto compMethod = parent->composite(&compTarget);
+
+    //If the parent & the only child have composition, we can't skip the parent....
+    if (compMethod != CompositeMethod::None && child->composite(nullptr) != CompositeMethod::None) return 0;
+
+    //propagate opacity
+    uint32_t opacity = parent->opacity();
+
+    if (opacity < 255) {
+        uint32_t tmp = (child->opacity() * opacity);
+        if (tmp > 0) tmp /= 255;
+        const_cast<Paint*>(child)->opacity(tmp);
```

Comment:
tmp = 0 if child->opacity() = 0, in such a case there is no point to calculate tmp and to set it as a child opacity.
what do you think about the code below? :
```
    if (opacity < 255) {
        if (auto tmp = child->opacity()) {
            tmp = (tmp * opacity) / 255;
            const_cast<Paint*>(child)->opacity(tmp);
        }
    }
```

Reply:
Either way, it needs to check opacity of parent to not dividing by zero.
it's ignorable...  I'd prefer simpler code than increase complexity  + 1

Reply:
@hermet you divide by 255, not by 0. 
even if tmp = 0 you can still divide it by 255. it will remain zero.
the thing is, that there is no point of doing anything if the child opacity is already zero.

Reply:
Yes anyway, divide 0 will be an issue. so your code is not proper.
and handling exceptions will code dirty and complex even we do there is no gain as you expected.